src/operator/l2_normalization-inl.h

```diff
@@ -235,6 +247,19 @@ class L2NormalizationProp : public OperatorProperty {
     return param_.__DICT__();
   }
 
+  bool InferType(std::vector<int> *in_type,
+                 std::vector<int> *out_type,
+                 std::vector<int> *aux_type) const override {
+    CHECK_EQ(in_type->size(), 1U);
+    int dtype = (*in_type)[0];
+    CHECK_NE(dtype, -1) << "Input must have specified type";
```

Comment:
Please use mutual inference instead of terminating the program.

Reply:
Done.